framework/src/bcs/LowerDIntegratedBC.C

```diff
@@ -34,15 +34,15 @@ LowerDIntegratedBC::LowerDIntegratedBC(const InputParameters & parameters)
     _test_lambda(_lowerd_var.phiLower())
 {
   const auto & lower_domains = _lowerd_var.activeSubdomains();
-  if (!lower_domains.count(Moose::BOUNDARY_SIDE_LOWERD_ID) && lower_domains.size() != 1)
-    paramError(
-        "lowerd_variable",
-        "Must be only defined on the subdomain BOUNDARY_SIDE_LOWERD_SUBDOMAIN subdomain that is "
-        "added by Mesh/build_all_side_lowerd_mesh=true");
-
-  if (_var.activeSubdomains().count(Moose::BOUNDARY_SIDE_LOWERD_ID))
-    paramError("variable",
-               "Must not be defined on the subdomain INTERNAL_SIDE_LOWERD_SUBDOMAIN subdomain");
+  for (const auto & id : _mesh.boundaryLowerDBlocks())
```

Comment:
this isnt quite the error we had before

we are basically forbidding the block restriction (on lower D blocks) of the variable. it must be defined on all lower D blocks

this should be fine for now

Reply:
Right. It is the current behavior. We can make it fancier in the future easily if someone requests.

Reply:
it is only the current behavior over our current set of tests. If I block restrict this variable to one of 2 lower D blocks , I will hit the error when I should not